lib/node/index.js

```diff
@@ -722,6 +722,7 @@ Request.prototype.request = function(){
   // request
   var req = this.req = mod.request(options);
   if ('HEAD' != options.method) req.setHeader('Accept-Encoding', 'gzip, deflate');
+  if ('DELETE' == options.method) req.setHeader('Content-Type', 0);
```

Comment:
What are you meaning by client? The client can sent the header he wants (see last test). Some web servers (like nginx) will deny a `DELETE` request when `Content-Length` is not set to `0`.